src/libraries/System.Net.NameResolution/src/System/Net/Dns.cs

```diff
@@ -525,28 +525,37 @@ private static Task GetHostEntryOrAddressesCoreAsync(string hostName, bool justR
 
                 asyncState = family == AddressFamily.Unspecified ? (object)ipAddress : new KeyValuePair<IPAddress, AddressFamily>(ipAddress, family);
             }
-            else if (NameResolutionPal.SupportsGetAddrInfoAsync)
+            else
             {
+                if (NameResolutionPal.SupportsGetAddrInfoAsync)
+                {
 #pragma warning disable CS0162 // Unreachable code detected -- SupportsGetAddrInfoAsync is a constant on *nix.
 
-                // If the OS supports it and 'hostName' is not an IP Address, resolve the name asynchronously
-                // instead of calling the synchronous version in the ThreadPool.
+                    // If the OS supports it and 'hostName' is not an IP Address, resolve the name asynchronously
+                    // instead of calling the synchronous version in the ThreadPool.
+                    // If it fails, we will fall-back to ThreadPool as well.
 
-                ValidateHostName(hostName);
+                    ValidateHostName(hostName);
 
-                if (NameResolutionTelemetry.Log.IsEnabled())
-                {
-                    return justAddresses
-                        ? (Task)GetAddrInfoWithTelemetryAsync<IPAddress[]>(hostName, justAddresses, family, cancellationToken)
-                        : (Task)GetAddrInfoWithTelemetryAsync<IPHostEntry>(hostName, justAddresses, family, cancellationToken);
-                }
-                else
-                {
-                    return NameResolutionPal.GetAddrInfoAsync(hostName, justAddresses, family, cancellationToken);
+                    Task t;
+                    bool success;
+                    if (NameResolutionTelemetry.Log.IsEnabled() || true)
```

Comment:
```suggestion
                    if (NameResolutionTelemetry.Log.IsEnabled())
```

Reply:
oops. forget this when I was forcing the telemetry path.